### PR TITLE
Adds support to read in string global attributes

### LINKED ISSUE #480: FMS2: get_global_attribute should support text type
**Describe the bug**
When updating to fms2 io, [get_global_attribute](https://github.com/NOAA-GFDL/FMS/blob/master/fms2_io/include/get_global_attribute.inc#L67), does not support an ```attributevalue``` containing text, and the model fails with an "unsupported type" error.  get_global_attribute is replacing [get_global_att_value](https://github.com/NOAA-GFDL/FMS/blob/master/fms/fms_io.F90#L7902) in fms which was able to handle text.

**To Reproduce**
I am seeing this when I run with my laurenchilutti/GFDL_atmos_cubed_sphere fork's [fms2io_part2](https://github.com/laurenchilutti/GFDL_atmos_cubed_sphere/tree/fms2io_part2) branch. 

**Expected behavior**
The model should run without "unsupported type" fatal error.

**System Environment**
Describe the system environment, include:
Running on Gaea with default modules loaded 

**Additional context**
You can see the output from my failed run at:
/lustre/f2/dev/Lauren.Chilutti/NEST/batch/fvgfs_201903_test/20160801.00Z.C48.nh.32bit.non-monoctl_ncep.res/rundir/fms_d2.out

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
This adds support to read in string global attributes. 
This adds a unit test that test the functionality in `get_global_attribute.inc` and `register_global_attribute.inc`

Fixes #480 

**How Has This Been Tested?**
Travis CI + a small unit test I created (attached here) in GAEA, intel18

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by uramirez8707** [comment]
@laurenchilutti Please try this with your test case
@menzel-gfdl Please take a look.

**Comment by laurenchilutti** [comment]
I've tested this with the changes in pull request #467 and the code runs as expected - no longer getting the unsupported type error.

File: fms2_io/include/get_global_attribute.inc [hunk 1]
```diff
@@ -32,8 +32,15 @@ subroutine get_global_attribute_0d(fileobj, &
                                              !! The broadcast will be done
                                              !! by default.
     integer :: err
+    character(len=1024), dimension(1) :: charbuf !< 1D Character buffer
```
Comment (bensonr) [inline fms2_io/include/get_global_attribute.inc:35]: Is there a reason charbuf needs to be defined as an array of length '1'?

Comment (menzel-gfdl) [reply fms2_io/include/get_global_attribute.inc:35]: I'm not sure there is a scalar string broadcast:

https://github.com/NOAA-GFDL/FMS/blob/94afa26669c1d1659f670cb739c0e9b4e51e7ece/mpp/include/mpp_comm_mpi.inc#L415

File: fms2_io/include/get_global_attribute.inc [hunk 2]
```diff
@@ -32,8 +32,15 @@ subroutine get_global_attribute_0d(fileobj, &
...
+
     if (fileobj%is_root) then
         select type(attribute_value)
+            type is (character(len=*))
+                err = nf90_get_att(fileobj%ncid, &
+                                   nf90_global, &
+                                   trim(attribute_name), &
+                                   charbuf(1))
```
Comment (thomas-robinson) [inline fms2_io/include/get_global_attribute.inc:43]: Will this always work with all compilers?  Do you need to pass in the length of the string with the string?

Comment (uramirez8707) [reply fms2_io/include/get_global_attribute.inc:43]: PGI support is going to be added in a seperate PR. This works for all **normal** compilers.

Comment (uramirez8707) [reply fms2_io/include/get_global_attribute.inc:43]: I will be adding the work around for PGI in a separate PR. 
@colingladueNOAA Therefore, there is no need to add the test in the configure we talked about in the meeting.
I think this is ready to go in.

**Pull request merged**